internal/datasources/sample_buckets.go

```diff
@@ -0,0 +1,214 @@
+package datasources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/bucket"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &SampleBuckets{}
+	_ datasource.DataSourceWithConfigure = &SampleBuckets{}
+)
+
+// Sample buckets is the bucket data source implementation.
+type SampleBuckets struct {
+	*providerschema.Data
+}
+
+// NewSampleBuckets is a helper function to simplify the provider implementation.
+func NewSampleBuckets() datasource.DataSource {
+	return &SampleBuckets{}
+}
+
+// Metadata returns the bucket data source type name.
+func (d *SampleBuckets) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_samplebuckets"
+}
+
+// Schema defines the schema for the bucket data source.
+func (s *SampleBuckets) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"organization_id": schema.StringAttribute{
+				Required: true,
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+			},
+			"cluster_id": schema.StringAttribute{
+				Required: true,
+			},
+			"data": schema.ListNestedAttribute{
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Computed: true,
+						},
+						"name": schema.StringAttribute{
+							Computed: true,
+						},
+						"organization_id": schema.StringAttribute{
+							Computed: true,
+						},
+						"project_id": schema.StringAttribute{
+							Computed: true,
+						},
+						"cluster_id": schema.StringAttribute{
+							Computed: true,
+						},
+						"type": schema.StringAttribute{
+							Optional: true,
```

Comment:
Is there any specific reason to include an optional type in the datasource field?

Reply:
Yes Sure, but test it after removing it.

Reply:
I've raised another [ticket](https://couchbasecloud.atlassian.net/browse/AV-74460) to sort out this and the different struct use case for the normal buckets